### PR TITLE
scap facts module

### PR DESCRIPTION
**Pull request by juliedavila**
##### SUMMARY
This a new module that allows for ansible to ingest the results.xml file generated by openscap-scanner's `oscap xccdf eval ...` command. The data is then stored as a fact.

Given that openscap, etc are generally always used in the context of audits etc, this facts module exposes the ability to cause the task to fail in the even certain score thresholds are not met either at the macro level or at the severity level. This is totally optional behavior and all thresholds are set to 0 by default.

Additionally, in order to make things easier to grok, a user can have the facts module only 'store' data for benchmarks that are passing, failing, not selected, not applicable, or not checked (or any combination).

##### ISSUE TYPE
New Module Pull Request

##### COMPONENT NAME
scap_facts

##### ANSIBLE VERSION
```
ansible 2.8.0.dev0 (oscap-facts eaaaeb2062) last updated 2018/11/02 09:50:17 (GMT -400)
  config file = None
  configured module search path = ['/Users/jonathand/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jonathand/Repos/ansible/lib/ansible
  executable location = /Users/jonathand/Repos/ansible/bin/ansible
  python version = 3.7.0 (default, Jul 23 2018, 20:22:55) [Clang 9.1.0 (clang-902.0.39.2)]
```

### CODE REVIEW

File: lib/ansible/modules/system/scap_facts.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+            result = res.find(self.ns + 'result').text
+            self.results[id]['result'] = result
+
+    def _organize_data(self, severity, data):
+        def percentage(part, whole):
+            return 100 * float(part) / float(whole)
+        raw = dict((k, v) for k, v in data.items() if v['severity'] == severity)
+        wanted = dict((k, v) for k, v in raw.items() if v['result'] in self.include_results)
+        fail = dict((k, v) for k, v in raw.items() if v['result'] == 'fail')
+        passing = dict((k, v) for k,v in raw.items() if v['result'] == 'pass')
```
Comment (samdoran) [inline lib/ansible/modules/system/scap_facts.py]:
```suggestion
        passing = dict((k, v) for k, v in raw.items() if v['result'] == 'pass')
```

**Comment by juliedavila** [comment]
related proposal https://github.com/ansible/proposals/issues/148

**Comment by mattclay** [comment]
CI failure in integration tests (traceback):

```
01:05 TASK [scap_facts : High Sev Criteria check] ************************************
01:06 An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: cannot concatenate 'str' and 'int' objects
01:06 fatal: [testhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/var/root/.ansible/tmp/ansible-tmp-1541171663.29-172248541979603/AnsiballZ_scap_facts.py\", line 113, in <module>\n    _ansiballz_main()\n  File \"/var/root/.ansible/tmp/ansible-tmp-1541171663.29-172248541979603/AnsiballZ_scap_facts.py\", line 105, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/var/root/.ansible/tmp/ansible-tmp-1541171663.29-172248541979603/AnsiballZ_scap_facts.py\", line 48, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_scap_facts_payload_UDiWH_/__main__.py\", line 354, in <module>\n  File \"/tmp/ansible_scap_facts_payload_UDiWH_/__main__.py\", line 332, in main\nTypeError: cannot concatenate 'str' and 'int' objects\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by gundalow** [comment]
dfed can we can close this

**Pull request closed without merging**